### DISCUSSION TITLE
environment variables with Kubernetes Template systemd Integration

### DISCUSSION THREAD

**Discussion by jguady** [containers/podman#18794] [Q&A]
Long title, hopefully somewhat self explanatory. 

So I'm trying to use podman to deploy our services/pod. I'm also trying to take advantage of the podman-kube@.service systemd template to manage my pod. 

However I'm also trying to set the http_proxy and https_proxy variables in the containers environment. For some reason the variables are not getting passed into the containers even if they are set in the host environment. 

Is there a recommendation for how to do this? I am fairly new to kubernetes but it sounds like I might need to create configmap? Ideally I could just pass it in with my .yml file. like ${http_proxy} but it seems to pick that value up literally.

**Comment by Luap99** [comment]
I have not checked if it works but if you set `http_proxy=true` in containers.conf it should pass the host proxy envs into the container.

Variables in the yaml are not supported see https://github.com/containers/podman/issues/15983

**Reply by jguady** [reply]
I didn't have luck with this at first but found the variables were being set in a dockerfile so I am retrying...

**Reply by jguady** [reply]
I'm not having any luck with the containers.conf file as suggested. It looks like variables are able to be passed by default as the docs say with podman run and using hashicorp/echo as a test container. 
`[]# podman inspect echos | grep proxy
                    "https_proxy=https://proxy.me:443",
[]# podman run -d --rm --name echos hashicorp/http-echo:latest -text "Hello"
`
However I am still unable to get my containers to pickup these variables from systemd template and the podman kube play. 

I'm open to more suggestions for things to try, in the meantime I'm checking to see if there are other reasons podman might not be passing these variables.

**Reply by Luap99** [reply]
If you just use `podman kube play` from the cli are the http proxy vars propagated there?

**Reply by jguady** [reply]
It looks like podman kube play does not pass the http proxy vars from the host to the container.

I took my test container and did a `podman generate kube` for it. with no proxy environment variables set. 
`# podman generate kube echos -f echo.yml`
Then I exported a proxy variable
`# export https_proxy=https://jay.proxy:443`
Then I used podman kube play on that file 
``` 
podman kube play echo.yml 
Pod:
cae5712befb92ce59b76301b296825a0376f49c23cd052bba431e0da4502d91c
Container:
36853f65ecf1b6fcd23540f9ef7ebb871bb20e7fe54874838a77d2413f7847ed 
```
Checked for proxy variable
``` 
podman inspect echos | grep proxy

``` 
no data

confirm env has proxy
```
# env | grep proxy
https_proxy=https://jay.proxy:443
```

So it seems podman kube play does not pass variables? or perhaps its the kube file being very literal in what it is setting. 

I was able to modify the podman-kube@.service and add a line 
`Environment='http_proxy=...'`
but this required re-creating the unit

**Reply by Luap99** [reply]
If `podman kube play` does not forward the proxy vars when it is set in containers.conf then this is a bug. It should work the same as podman run. containers.conf options should apply to all containers regardless how they are created.
